src/libraries/System.Memory/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -476,10 +477,31 @@ private static unsafe void Avx2Decode(ref byte* srcBytes, ref byte* destBytes, b
             destBytes = dest;
         }
 
+        // This can be replaced once https://github.com/dotnet/runtime/issues/63331 is implemented.
+        // Note that for performance reasons this is not functionally identical on SSE3 and Arm64. On both, the input
+        // right is truncated to 4bits per lane. In addition on SSE3, the shuffle will return 0 for each input in right
+        // where the top bit is set. On Arm64 the top bit will be ignored.
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static unsafe void Ssse3Decode(ref byte* srcBytes, ref byte* destBytes, byte* srcEnd, int sourceLength, int destLength, byte* srcStart, byte* destStart)
+        private static Vector128<byte> SimdShuffle(Vector128<byte> left, Vector128<byte> right, Vector128<byte> maskf)
```

Comment:
@tannergooding : How does your Vector128.Shuffle() compare to this code? On Arm64 does it just resolve down to a VectorTableLookup with And ?

On SSE3, the hardware shuffle instruction will "return 0 for each input in right where the top bit is set".  Is that difference being taken in account for Arm64?

Reply:
`Vector128.Shuffle` currently works best with a constant input (that is where the mask is `Vector128.Create(....)`). 

It behaves such that any out of range index causes the corresponding element to be zeroed (so behaves more like `VectorTableLookup` and on x86/x64 it will have to `compare`, then `or` to have the right `pshufb` behavior).

Reply:
Ok, that makes sense.
if I change my patch from SSE3.shuffle() to Vector128.Shuffle(), then it's going to be slower on X86 ? If so what do you recommend using (Vector128 or the code I have)?

Reply:
Its possible that it'd have an impact yes. If that's a concern having a small helper that does `if (Sse3.IsSupported) { Sse3.Shuffle } else if (AdvSimd.IsSupported) { AdvSimd.VectorTableLookup } else { throw new NotSupportedException(); }` is a reasonable and simple workaround to keep things shared but still with a platform specific optimization.

Reply:
Ok, I'll try this out with the performance tests, and see what the difference is.

Reply:
That's a _lot_ slower when I use Vector128.Shuffle(left, Vector128.BitwiseAnd(right, maskf));
Will keep with the code as it is then.

Reply:
I'd expect its slower because `maskf` isn't being detected as a constant. I'm working on the PR (should be up by end of week) that handles non-constants but it will still be 3-5 instructions instead of just 1 instruction.